Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -1351,7 +1357,12 @@ bool qSlicerExtensionsManagerModel::installExtension(
       // Install dependencies
       foreach (const ExtensionMetadataType& dependency, dependenciesMetadata)
         {
-        this->downloadAndInstallExtension(dependency.value("extension_id").toString());
+        bool res = this->downloadAndInstallExtension(dependency.value("extension_id").toString());
+        if( !res )
+          {
+          QString msg = QString("<p>Error while installing %1</p><ul>").arg(dependency.value("extensionname").toString());
+          QMessageBox::warning(0, "Error installing dependencies", msg);
```

Comment:
Instead of showing a `QMessageBox` for each dependency that failed to be installed. Could you instead accumulate the error in a `QStringList` and then call `this->critical(errors.join("\n"))` outside of the loop ? If not, the user will have to acknowledge the popup for each dependencies ...